analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/DeconstructionAssignment.cs

```diff
@@ -0,0 +1,68 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class DeconstructionAssignment : SimpleProcessor<IDeconstructionAssignmentOperationWrapper>
+{
+    protected override IDeconstructionAssignmentOperationWrapper Convert(IOperation operation) =>
+        IDeconstructionAssignmentOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState Process(SymbolicContext context, IDeconstructionAssignmentOperationWrapper assignment)
+    {
+        if (assignment.Target.Kind != OperationKindEx.Tuple)
+        {
+            return context.State;
+        }
+
+        var targetTuple = assignment.Target.ToTuple();
+        var newState = context.State;
+        if (assignment.Value.AsTuple() is { } rightSideTuple)
```

Comment:
This likely misses nested tuples. See 
https://github.com/SonarSource/sonar-dotnet/blob/c4fe953456f49d0b9321b0ad0f1a9903be8f6e14/analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs#L69-L81

and

https://github.com/SonarSource/sonar-dotnet/blob/c4fe953456f49d0b9321b0ad0f1a9903be8f6e14/analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AssignmentExpressionSyntaxExtensionsTests.cs#L30

for details how we are doing this on a syntax level. We need something equivalent for IOperation.

Reply:
Thanks. Added test cases for nested scenarios.